lib/algoliasearch-rails.rb

```diff
@@ -399,7 +399,8 @@ def algoliasearch(options = {}, &block)
         raise ArgumentError.new("Cannot use a enqueue if the `synchronous` option if set") if options[:synchronous]
         proc = if options[:enqueue] == true
           Proc.new do |record, remove|
-            AlgoliaJob.perform_later(record, remove ? 'algolia_remove_from_index!' : 'algolia_index!')
+            record_or_object_id = remove ? algolia_object_id_of(record) : record
```

Comment:
@dsgh @DevinCodes I think this line needs to be more general. If a single model instance is included in two indexes, and has a unique `objectID` in each of those indexes, this will not remove the record from each Algolia index in which it's included. 

To remove the instance from each index in which it's included, we need to specify the Algolia index options for each relevant index in this call to `algolia_object_id_of`. i.e. this should probably be an array of `objectID` values, one per relevant index

Reply:
For what it's worth, we have a pretty complex setup in which several models are in several indices, and many of those indices include multiple models. Some of the models also use custom `objectID` values in indexes (so that e.g. Message=3 and User=3 will not have colliding `objectID`s in the indices in which both are present.

When a record needs to be deleted, we find each of the indexes in which the record is included, and find the name of each of those indexes. Then we send that 2D array of [[object_id, index_name], ...] to our ActiveJob, from which we can retrieve the relevant index with `index = Algolia::Index.new(index_name)` and then remove the object with `index.delete_object(object_id)`. This ensures that objects with custom object ids that are included in several indices are properly removed from each when the model instance is destroyed